lib/cb/utils/validator.rb

```diff
@@ -23,8 +23,19 @@ def validate(response)
 
       def raise_response_code_errors(response)
         code = response.code rescue nil
-        fail Cb::ServiceUnavailableError if code == 503 || simulate_auth_outage?
-        fail Cb::UnauthorizedError if code == 401
+        fail_with_error_details(response, Cb::ServiceUnavailableError) if code == 503 || simulate_auth_outage?
+        fail_with_error_details(response, Cb::UnauthorizedError) if code == 401 || code == 403
+        fail_with_error_details(response, Cb::DocumentNotFoundError) if code == 404
+        fail_with_error_details(response, Cb::BadRequestError) if code && code >= 400 && code < 500
+        fail_with_error_details(response, Cb::ServerError) if code && code >= 500
```

Comment:
What do you think about switching the `UnauthorizedError` line to `[401, 403].include?(code)` instead? Could also argue for moving the `if code` that is on the last couple and instead wrap the set of 5 lines in an if block for that.